fedimint-client/src/lib.rs

```diff
@@ -232,6 +232,28 @@ impl Event for TxRejectedEvent {
     const KIND: EventKind = EventKind::from_static("tx-rejected");
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct ModuleRecoveryStarted {
+    module_id: ModuleInstanceId,
+}
+
+impl Event for ModuleRecoveryStarted {
+    const MODULE: Option<ModuleKind> = None;
+
+    const KIND: EventKind = EventKind::from_static("module-recovery-started");
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ModuleRecoveryCompleted {
+    module_id: ModuleInstanceId,
+}
+
+impl Event for ModuleRecoveryCompleted {
+    const MODULE: Option<ModuleKind> = None;
+
+    const KIND: EventKind = EventKind::from_static("module-recovery-completed");
+}
+
```

Comment:
Would  it be worth just having one event type that includes progress? From that one could gauge if it just started, is ongoing or finished.

Reply:
I want to have a precise and reliable single start & end for each module recovery, so the subscribers can reliably use it for controlling their logic. Progress could be another event, though I'm a bit worried about these being too spammy.